src/fondant/runner.py

```diff
@@ -26,3 +29,30 @@ def run(cls, input_spec: str, *args, **kwargs):
         ]
 
         subprocess.call(cmd)  # nosec
+
+
+class KubeflowRunner(Runner):
+    @abstractmethod
+    def _resolve_imports(self):
+        """Resolve imports for the Kubeflow compiler."""
+        try:
+            import kfp
+        except ImportError:
+            raise ImportError(
+                "You need to install kfp to use the Kubeflow compiler, "
+                / "you can install it with `poetry install --extras kfp`"
+            )
+
+    def run(cls, input_spec: str, host: str, *args, **kwargs):
+        """Run a kubeflow pipeline."""
+        cls._resolve_imports()
+        client = kfp.Client(host=host)
```

Comment:
This probably doesn't change a lot to the current functionality, but it would make more sense to me to instantiate this in `__init__()`.